### ISSUE TITLE
Remove leading spaces in coderefs

### ISSUE THREAD

**Issue by xephon2** [dita-ot/dita-ot#2907]
I really like the coderef feature to pick certain parts of config files for explanation.

```xml
<codeblock outputclass="language-xml">
  <coderef href="DBService_App.config#token=snippet-START,snippet-END"/>
</codeblock>
```

**Problem**

It would be a great enhancement, if the indentation of these blocks would be fixed. Some codeblocks have very long lines, that look messy in the out. Think of this block:

```xml
            <property name="dataStorageConfiguration">
                <bean class="org.apache.ignite.configuration.DataStorageConfiguration">
                    <property name="defaultDataRegionConfiguration">
                        <bean class="org.apache.ignite.configuration.DataRegionConfiguration">
                            <property name="name"
                                      value="ContactGatewayDefault"/>
                            <property name="initialSize"
                                      value="#{1L * 1024 * 1024 * 1024}"/>
                            <property name="maxSize"
                                      value="#{4L * 1024 * 1024 * 1024}"/>
                            <property name="metricsEnabled"
                                      value = "true"/>
                            <property name="persistenceEnabled"
                                      value="false"/>
                        </bean>
                    </property>
                </bean>
            </property>
```

To make this block look good, you have to manually remove the leading spaces to make it look like this:

```xml
<property name="dataStorageConfiguration">
    <bean class="org.apache.ignite.configuration.DataStorageConfiguration">
        <property name="defaultDataRegionConfiguration">
            <bean class="org.apache.ignite.configuration.DataRegionConfiguration">
                <property name="name"
                          value="ContactGatewayDefault"/>
                <property name="initialSize"
                          value="#{1L * 1024 * 1024 * 1024}"/>
                <property name="maxSize"
                          value="#{4L * 1024 * 1024 * 1024}"/>
                <property name="metricsEnabled"
                          value = "true"/>
                <property name="persistenceEnabled"
                          value="false"/>
            </bean>
        </property>
    </bean>
</property>
```

That is boring and expensive work. It makes the file looking good for the tech writer, but ugly for the dev, because it destroys the indentation of the original file.

**Solution**
All referenced lines of a code block need to be parsed. If all lines have leading spaces, find the line with the fewest spaces, count the spaces and remove the amount of spaces from all referenced lines.

**Comment by jelovirt** [comment]
This is something I've been planning to implement for ages. I've been thinking about adding support for magic `dita-ot:trim-whitespace` `@outputclass` value that would trim whitespace as described above. This could also be used in `<codeblock>`, not sure if it should also be supported on `<coderef>`.

**Comment by infotexture** [comment]
Implemented in #2953 for 3.1 with b768808 and 8f16633.
